launch/model_endpoint.py

```diff
@@ -1,18 +1,36 @@
 import concurrent.futures
 import uuid
 from collections import Counter
+from dataclasses import dataclass
 from typing import Dict, Optional, Sequence
 
+from dataclasses_json import Undefined, dataclass_json
+
 from launch.request_validation import validate_task_request
 
 TASK_PENDING_STATE = "PENDING"
 TASK_SUCCESS_STATE = "SUCCESS"
 TASK_FAILURE_STATE = "FAILURE"
 
 
+@dataclass_json(undefined=Undefined.EXCLUDE)
+@dataclass
+class ModelEndpoint:
```

Comment:
I think it's a bit confusing to have both `ModelEndpoint` and `Endpoint` - seems like this class is more of an "endpoint params" that represents the common initialization fields for either endpoint type. But we can deal with this later - I think the overall client library needs a facelift before we go live anyway.